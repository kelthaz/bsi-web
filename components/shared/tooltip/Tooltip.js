import React from 'react';
import PropTypes from 'prop-types';
import styles from './tooltip.module.scss';
import SvgInformation from '../../svgs/SvgInformation';

const Tooltip = ({ message, position }) => (
  <span className={styles.tooltip}>
    <SvgInformation />
    <span
      className={`${styles.tooltiptext} ${position === 'bottom' ? styles['position-bottom'] : styles['position-top']}`}
    >
      {message}
    </span>
  </span>
);

Tooltip.propTypes = {
  message: PropTypes.string.isRequired,
  position: PropTypes.string,
};

Tooltip.defaultProps = {
  position: 'botttom',
};
export default Tooltip;
