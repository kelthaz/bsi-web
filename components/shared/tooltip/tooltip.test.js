import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import Tooltip from './Tooltip';

describe('Pruebas en el componente Tooltip', () => {
  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const msg = 'Esta es tu edad';
    // act
    const wrapper = shallow(<Tooltip message={msg} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de mostrarse el mensaje', () => {
    // arrange
    const msg = 'Esta es tu edad';
    const wrapper = shallow(<Tooltip message={msg} />);
    // act
    const span = wrapper.find('span').at(1).text().trim();
    // assert
    expect(span).toBe(msg);
  });
  test('Debe de mostrarse el mensaje en la parte inferior', () => {
    // arrange
    const msg = 'Esta es tu edad';
    const wrapper = shallow(<Tooltip message={msg} position="bottom" />);
    // act
    const span = wrapper.find('span').at(1).text().trim();
    // assert
    expect(span).toBe(msg);
  });
});
