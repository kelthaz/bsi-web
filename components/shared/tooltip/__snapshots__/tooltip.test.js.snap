// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Tooltip Debe de mostrarse correctamente el componente 1`] = `
<Fragment>
  <span
    className="tooltip"
  >
    Â 
    <SvgInformation />
  </span>
  <span
    className="tooltiptext position-top"
  >
    Esta es tu edad
  </span>
</Fragment>
`;
